### DISCUSSION TITLE
browser show error cors after OPTIONS pre-flight requests active

### DISCUSSION THREAD

**Discussion by prasetyo531** [mockoon/mockoon#1481] [Help]
im working with testing frontend and using url mockoon as my .env, have folllow below step but cors still occured

https://mockoon.com/docs/latest/server-configuration/cors/
<img width="1467" alt="Screen Shot 2024-08-08 at 11 55 03" src="https://github.com/user-attachments/assets/ee244b53-2b96-43e5-9f83-5e629832109e">

error log

`'http://localhost:6001' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header has a value 'http://localhost:3000' that is not equal to the supplied origin.`

http://localhost:6001 is my local build
<img width="1008" alt="Screen Shot 2024-08-08 at 11 55 52" src="https://github.com/user-attachments/assets/7be415c6-dde1-475a-86bc-902ca3533d65">

checking log in mockoon somehow only got request OPTIONS 
<img width="1474" alt="Screen Shot 2024-08-08 at 12 06 02" src="https://github.com/user-attachments/assets/341ae018-cf4a-4bfd-b4ed-a933d4e19750">

but request using postman is fine
<img width="1258" alt="Screen Shot 2024-08-08 at 12 06 50" src="https://github.com/user-attachments/assets/34236177-729a-4bad-9c62-c42aedafbb06">

Mockoon version:
8.0

OS / OS version:
Mac OS

**Comment by 255kb** [comment]
Mockoon will still use your environment headers to answer OPTIONS pre-flight requests.
You should check the value of the `Access-Control-Allow-Origin` header: 
![image](https://github.com/user-attachments/assets/cbbc29aa-22ff-49f4-a0e0-896008c476a0)

**Reply by prasetyo531** [reply]
hi thanks, forgot to mention it, 
but i still got cors 
<img width="1417" alt="Screen Shot 2024-08-08 at 13 07 10" src="https://github.com/user-attachments/assets/a8a14607-ba62-4bbe-a3fc-cebf1be24703">

**Reply by 255kb** [reply]
That's odd. Answers to a preflight OPTIONS should be `Access-Control-Allow-Origin: *` or the value of the environment header (I see `*` in your case too). I don't know where the `http://localhost:3000` is coming from. 

Please check if you don't have an OPTIONS route declared that is returning the wrong headers. 
Without having access to the full setup I'm not sure where this comes from.

**Reply by geekmedium** [reply]
hi @255kb its fixed for previous problem, turns out i just need add `pragma` in Access-Control-Allow-Headers.
